spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationConfig.scala

```diff
@@ -0,0 +1,169 @@
+      }).text("Specifies if the mainframe file is ASCII text file"),
+
+      opt[String]("cobol-encoding").optional().action((value, config) => {
+        val newOptions = config.cobolOptions match {
+          case Some(a) => Some(a.copy(encoding = Option(value)))
+          case None => Some(CobolOptions(encoding = Option(value)))
+        }
+        config.withCobolOptions(newOptions)
+      }).text("Specify encoding of mainframe files (ascii or ebcdic)"),
+
+      opt[String]("cobol-trimming-policy").optional().action((value, config) => {
+        val newOptions = config.cobolOptions match {
+          case Some(a) => Some(a.copy(trimmingPolicy = Option(value)))
+          case None => Some(CobolOptions(trimmingPolicy = Option(value)))
+        }
+        config.withCobolOptions(newOptions)
+      }).text("Specify string trimming policy for mainframe files (none, left, right, both)"),
+
+      opt[String]("debug-set-raw-path").optional().hidden().action((value, config) =>
+        config.withRawPathOverride(Some(value)))
+        .text("override the path of the raw data (used internally for performance tests)"),
+
+      checkConfig(config => {
+        def foundCsvField(config: R) = config match {
+          case _ if config.csvDelimiter.isDefined => Some("--delimiter")
+          case _ if config.csvEscape.isDefined => Some("--escape")
+          case _ if config.csvHeader.contains(true) => Some("--header")
+          case _ if config.csvQuote.isDefined => Some("--quote")
+          case _ => None
+        }
+
+        def foundCobolField(cobolOptions: CobolOptions) = cobolOptions match {
+          case _ if cobolOptions.copybook != "" => Some("--copybook")
+          case _ if cobolOptions.encoding.isDefined => Some("--cobol-encoding")
+          case _ if cobolOptions.isXcom => Some("--is-xcom")
+          case _ if cobolOptions.isText => Some("--is-text")
+          case _ if cobolOptions.trimmingPolicy.isDefined => Some("--cobol-trimming-policy")
+          case _ => None
+        }
+
+
+        if (!List("xml", "csv", "json", "cobol").contains(config.rawFormat.toLowerCase) && config.charset.isDefined)
+          failure("The --charset option is supported only for CSV, JSON, XML and COBOL")
+        else if (config.rowTag.isDefined && !config.rawFormat.equalsIgnoreCase("xml"))
+          failure("The --row-tag option is supported only for XML raw data format")
+        else if (foundCsvField(config).isDefined && !config.rawFormat.equalsIgnoreCase("csv")) {
+          foundCsvField(config) match {
+            case Some("header") => failure("The --header option is supported only for CSV ")
+            case Some(field) => failure(s"The $field option is supported only for CSV raw data format")
+            case None =>
+              val cobolOptions = config.cobolOptions
+              if (cobolOptions.isDefined && !config.rawFormat.equalsIgnoreCase("cobol")) {
+                val field = foundCobolField(cobolOptions.get)
```

Comment:
1. Why the cobol branch is executed only if `foundCsvField(config).isDefined && !config.rawFormat.equalsIgnoreCase("csv")` holds?

2. If the validation is postponed to this point and `cobolOptions` is defined, you can get the list of *all* non-empty cobol options and include the list of options supported only in `cobol`.
The way it is currently implemented it seems it takes an arbitrary non-empty cobol option and fails on it.